xcodeproj/internal/xcode_targets.bzl

```diff
@@ -308,26 +322,51 @@ def _merge_xcode_target(*, src, dest):
         extensions = dest._extensions,
         app_clips = dest._app_clips,
         dependencies = depset(
-            transitive = [dest._dependencies, src._dependencies],
+            transitive = [dest._dependencies] + [src._dependencies for src in srcs],
         ),
         transitive_dependencies = dest.transitive_dependencies,
-        outputs = _merge_xcode_target_outputs(
-            src = src.outputs,
+        outputs = _merge_xcode_targets_outputs(
+            src_outputs = [
+                src.outputs
+                for src in srcs
+            ],
             dest = dest.outputs,
         ),
         lldb_context = dest.lldb_context,
         xcode_required_targets = dest.xcode_required_targets,
     )
 
-def _merge_xcode_target_inputs(*, src, dest):
+def _merge_xcode_targets_inputs(*, src_inputs, dest):
+    pchs = [src.pch for src in src_inputs if src.pch]
```

Comment:
Same comment, but this should only come from the non-swift target. I would pass on two inputs, one for each of the two targets.